src/page/text.rs

```diff
@@ -8,43 +8,46 @@
 // At your choosing (See accompanying files LICENSE_APACHE_2_0.txt,
 // LICENSE_MIT.txt and LICENSE_BOOST_1_0.txt).
 
-
 /// Text formatting
 pub enum Format {
     /// Paragraph
-    Paragraph,
+    Paragraph(Box<Self>),
     /// Code block
-    Code,
+    Code(Box<Self>),
     /// Quote block begin
-    QuoteIn,
+    QuoteIn(Box<Self>),
     /// Quote block end
-    QuoteOut,
+    QuoteOut(Box<Self>),
     /// H1, usually title of document
-    Title,
+    Title(Box<Self>),
     /// H2, section 1
-    Section,
+    Section(Box<Self>),
     /// H3, section 1.1
-    Header,
+    Header(Box<Self>),
     /// H4, section 1.1.1
-    Subheader,
+    Subheader(Box<Self>),
     /// Label for following user entry
-    Label,
+    Label(Box<Self>),
+    /// End the chain of formatting
+    ///
+    /// Should not be used on it's own.
+    None,
 }
```

Comment:
There's a reason I didn't write it like this; It's more space efficient to store `Format` in a `Vec` - Notice all variants point to another variant.  This can be encoded by a simple rule - the next element is the successor of the previous.

The variant types allow a "tree" (rendering showed after):

```rust
vec![
    Text::Format(Format::Paragraph), // Set style to paragraph
    Text::Strong(true), // Bold for English text
    Text::Span("Some bold text. "),
    Text::Emphasis(true), // Italic for English text
    Text::Span("Italic/Bold text within the bold text. "),
    Text::Emphasis(false), // Return to bold for English text
    Text::Strong(false), // Regular for English text
    Text::Span("Last sentence of a paragraph\n"),
    Text::Span("Next paragraph"),
    Text::Format(Format::Section)
    Text::Span("Next section"),
]
```

**some bold text. _Italic/Bold text within the bold text_.** Last sentence of a paragraph

Next paragraph

## Next section

Reply:
Hmm, this is a good point. I personally think the API of chaining would make more sense, but it's up to you.

Reply:
Chaining is essentially implementing a linked list, which is almost always not a good idea in Rust.  It might be able to be made cleaner with a builder but I'm not sure.

Reply:
Builders do tend to be quite clean @adaburton What's your opinion?